base_ical/controllers/main.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2023 Hunki Enterprises BV
+# Copyright 2024 initOS GmbH
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3.0)
+
+import logging
+
+import werkzeug.wrappers
+
+from odoo import http
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+
+class ICalController(http.Controller):
+    @http.route(
+        "/base_ical/<int:calendar_id>",
+        auth="none",
+        csrf=False,
+        methods=["GET"],
+        type="http",
+    )
+    def get_ical(self, calendar_id, access_token=None):
+        if not access_token:
+            raise request.not_found()
+
+        user_id = request.env["res.users.apikeys"]._check_credentials(
+            scope=f"odoo.plugin.ical.{calendar_id}",
+            key=access_token,
+        )
+        if not user_id:
+            raise request.not_found()
+
+        calendar = (
+            http.request.env["base.ical"]
+            .sudo()
```

Comment:
Given that users have read access to base.ical model per the security csv, replace `sudo()` with `with_user(user_id)` from below?